src/AppInstallerCLICore/Commands/UpdateCommand.cpp

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "UpdateCommand.h"
+#include "Workflows/CompletionFlow.h"
+#include "Workflows/InstallFlow.h"
+#include "Workflows/UpdateFlow.h"
+#include "Workflows/WorkflowBase.h"
+#include "Resources.h"
+
+using namespace AppInstaller::CLI::Execution;
+using namespace AppInstaller::Manifest;
+using namespace AppInstaller::CLI::Workflow;
+
+namespace AppInstaller::CLI
+{
+    std::vector<Argument> UpdateCommand::GetArguments() const
+    {
+        return {
+            Argument::ForType(Args::Type::Query),
+            Argument::ForType(Args::Type::Manifest),
+            Argument::ForType(Args::Type::Id),
+            Argument::ForType(Args::Type::Name),
+            Argument::ForType(Args::Type::Moniker),
+            Argument::ForType(Args::Type::Version),
+            Argument::ForType(Args::Type::Channel),
+            Argument::ForType(Args::Type::Source),
+            Argument::ForType(Args::Type::Exact),
+            Argument::ForType(Args::Type::Interactive),
+            Argument::ForType(Args::Type::Silent),
+            Argument::ForType(Args::Type::Language),
+            Argument::ForType(Args::Type::Log),
+            Argument::ForType(Args::Type::Override),
+            Argument::ForType(Args::Type::InstallLocation),
+            Argument{ "force", Argument::NoAlias, Args::Type::Force, Resource::String::InstallForceArgumentDescription, ArgumentType::Flag },
+            Argument{ "all", Argument::NoAlias, Args::Type::All, Resource::String::UpdateAllArgumentDescription, ArgumentType::Flag },
+        };
+    }
+
+    Resource::LocString UpdateCommand::ShortDescription() const
+    {
+        return { Resource::String::UpdateCommandShortDescription };
+    }
+
+    Resource::LocString UpdateCommand::LongDescription() const
+    {
+        return { Resource::String::UpdateCommandLongDescription };
+    }
+
+    void UpdateCommand::Complete(Execution::Context&, Execution::Args::Type) const
+    {
+        // TODO: Should be done similar to list completion
+    }
+
+    std::string UpdateCommand::HelpLink() const
+    {
+        // TODO: point to correct location
+        return "https://aka.ms/winget-command-upgrade";
+    }
+
+    void UpdateCommand::ExecuteInternal(Execution::Context& context) const
+    {
+        context.Get<Execution::Data::ContextFlag>() |= Execution::ContextFlag::InstallerExecutionUseUpdate;
+
+        context <<
+            OpenSource <<
+            GetCompositeSourceFromInstalledAndAvailable;
+
+        if (context.Args.Empty())

```

Comment:
>if (context.Args.Empty()) [](start = 8, length = 25)

I think this might need an args validation function.

Reply:
Not Empty, but the command.

---
In reply to: [503511290](https://github.com/microsoft/winget-cli/pull/588#discussion_r503511290) [](ancestors = 503511290)